src/Nethermind/Nethermind.State/WorldStateManager.cs

```diff
@@ -10,19 +10,45 @@
 using Nethermind.Logging;
 using Nethermind.State.Healing;
 using Nethermind.State.Snap;
+using Nethermind.State.SnapServer;
 using Nethermind.Synchronization.FastSync;
+using Nethermind.Trie;
 using Nethermind.Trie.Pruning;
 
 namespace Nethermind.State;
 
-public class WorldStateManager(
-    IWorldState worldState,
-    ITrieStore trieStore,
-    IDbProvider dbProvider,
-    ILogManager logManager,
-    IProcessExitSource? processExitSource = null)
-    : ReadOnlyWorldStateManager(dbProvider, trieStore.AsReadOnly(), logManager, processExitSource)
+public class WorldStateManager : IWorldStateManager
 {
+    private readonly IWorldState _worldState;
+    private readonly ITrieStore _trieStore;
+    private readonly IReadOnlyTrieStore _readOnlyTrieStore;
+    private readonly ILogManager _logManager;
+    private readonly ReadOnlyDb _readaOnlyCodeCb;
+    private readonly IDbProvider _dbProvider;
+    private readonly BlockingVerifyTrie? _blockingVerifyTrie;
+
+    public WorldStateManager(IWorldState worldState,
+        ITrieStore trieStore,
+        IDbProvider dbProvider,
+        ILogManager logManager,
+        IProcessExitSource? processExitSource = null)
+    {
+        _dbProvider = dbProvider;
+        _worldState = worldState;
+        _trieStore = trieStore;
+        _readOnlyTrieStore = trieStore.AsReadOnly();
+        _logManager = logManager;
+
+        IReadOnlyDbProvider readOnlyDbProvider = dbProvider.AsReadOnly(false);
+        _readaOnlyCodeCb = readOnlyDbProvider.GetDb<IDb>(DbNames.Code).AsReadOnly(true);
+        GlobalStateReader = new StateReader(_readOnlyTrieStore, _readaOnlyCodeCb, _logManager);
+
+        if (processExitSource is not null)
+        {
+            _blockingVerifyTrie = new BlockingVerifyTrie(trieStore, GlobalStateReader, _readaOnlyCodeCb!, processExitSource!, logManager);
+        }
```

Comment:
It is bit weird that it depends on if exitsource is passed

Reply:
Yea, it should have been just one `void VerifyTrie(CancellationToken cancellation)` and the exit source is handled outside.

Reply:
Moved the exit source logic out.